### ISSUE TITLE
Update to use new report rate API.

### ISSUE THREAD

**Issue by Hjdskes** [libratbag/piper#189]
There might be something we can use there...

**Comment by whot** [comment]
I think we can largely leave it the same. most devices provide 500/1000, so all we have to do is check if those are there. If not, we can pick two other values that are provided or remove the UI widget altogether.

**Comment by phomes** [comment]
I think that few users understand what the option does and which value would be right for them. Perhaps we can make the choice easier by explaining the effect instead?

There are two reasons why you would not have the report rate at max. Allegedly some computers cannot handle a report rate of 1000. Lowering it would then be for backwards compatibility. The other reason is power. A wireless mouse will drain its battery faster at a higher report rate. It might also have an impact on a laptop too. Haswell and newer models can go to deeper sleep only if there is no activity on the PCH. (I can't test as my laptop is too old. Should also be tested with Hans' med_power_with_dipm or stricter policy to let the PCH rest in the first place).

I see this as an all or nothing option. Either you want it at min "power efficiency" for normal use or max "best precision" when gaming.

**Comment by Hjdskes** [comment]
@whot In any case, we'll need to either check that the 500/1000 options are there or pick two new values that are available instead, so there is something to be done.

It seems though that @phomes is hinting towards either removing this option altogether or only exposing the min and max value under the names "power efficiency" or "best precision". Am I right in concluding this from your comment @phomes? And if so, what would be the best solution here? Input from @whot @bentiss @phomes and others is appreciated :)

**Comment by whot** [comment]
I would start with the simple check they're there first, though technically that requires https://github.com/libratbag/libratbag/issues/406 to be fixed first too.

I'm not sure about renaming them though. Having a label next to it or a tooltip that explains it sounds good, but in the end our users are the ones that bought a gaming mouse with lots of toggles. We should assume that they roughly know what these are doing.

**Comment by Hjdskes** [comment]
> I would start with the simple check they're there first, though technically that requires libratbag/libratbag#406 to be fixed first too.

What is the progress on libratbag/libratbag#406? I'd prefer to work against something that is working, or I risk introducing features that rely on unimplemented work in libratbag.

> Having a label next to it or a tooltip that explains it sounds good, but in the end our users are the ones that bought a gaming mouse with lots of toggles. We should assume that they roughly know what these are doing.

Right, that is understandable. Once I get started on this (probably when the above libratbag issue is fixed) I will experiment with both a label and a tooltip to see what works best.

**Comment by whot** [comment]
progress: everyone is side-tracked ;) it'd be a simple enough change if you want to add it to libratbag, at least the basic flag.

**Comment by AlexFolland** [comment]
I'm not sure if this is the right place for this, but I'm commenting here to mention that my Logitech G700s mouse's report rate is set to 333 Hz in Logitech Gaming Software for Windows and 500 Hz in Manjaro, automatically loaded by and configured by piper.

I find 333 Hz to be a good neutral value for both performance and battery life.  With the minimum being 500 Hz in piper, the battery should theoretically be being consumed 50% faster in Linux than in Windows with the same mouse ((500-333)/333) is roughly 50%), and that does seem to be the case in my practical testing this week after switching to Manjaro.  In Windows with 333 Hz, I could leave my mouse unplugged during my entire daily computing session, which includes working from home, and only had to plug it in before sleeping, as that's when it became very low.  In Manjaro with this 500 Hz report rate, it tends to run critically low on battery about 2/3 through my session requiring me to plug it in at that time.  So, this setting not being available is affecting my daily activities.

I have replaced the previous 2450 mAh battery with a new 2450 mAh battery to ensure my mouse is getting the most battery life that it can.  It's quite possible that this mouse is not optimized well for battery life, but that should be the same regardless of operating system.

This issue for me is not critical, but I wanted to let you know about it as a user of piper.

I honestly haven't investigated bypassing piper and interfacing with libratbag directly via the terminal.  I'll do that when I have some time.

Thank you for piper.  It's awesome and it's one of my favorite tools that helped me switch from Windows to Manjaro over the weekend.

**Comment by staticssleever668** [comment]
If I understand the issue correctly, it meant following:
- Showing other than 500 and 1000hz report rates - we now also display 125 and 250 and disable buttons for unsupported report rates
- Use the per-profile report rate API instead of per-resolution - this has been like that for a while
So, closing it as fixed

**Comment by AlexFolland** [comment]
Is 333 Hz supported?  It is available in Logitech's tool for Windows.

**Comment by staticssleever668** [comment]
@AlexFolland, not through Piper, it only shows 125, 250, 500 and 1000 (which are the most popular report rates), but you can do it through ratbagctl - the command line utility for libratbag: `ratbagctl Logitech rate set 333`. (it was impossible at the time you asked the question as ratbagctl incorrectly rounded the numbers, and 333 would become 300).
By the way, if you have access to it, could you show me a picture of Logitech's tool's interface for changing report rates?

**Comment by AlexFolland** [comment]
Unfortunately I don't have access to Logitech's tool any more right now, as I have purged Windows from my systems.  I believe the tool in question is called "Logitech Gaming Software", and may have since been replaced by a different, more limited tool.  If this uncommon report rate can be set with the `ratbagctl` command line you've mentioned, then I am no longer personally concerned about this issue.  Thanks for your response and your effort improving the ease of configurability of the Linux desktop.
